### ISSUE TITLE
Same as JCS / RFC 8785?

### ISSUE THREAD

**Issue by joepio** [gelvinp/rs-serde_canonical_json#1]
Hi there! Thanks for making this :)

Does this implement [JCS / RDF 8785](https://www.rfc-editor.org/rfc/rfc8785)? 

I didn't see this mentioned in this repo, but looking at the implementation it looks pretty close.

**Comment by gelvinp** [comment]
Hi there,

No, this crate does not implement RDF 8785, and appears to be fundamentally incompatible in a few ways. That being said, depending on your use case, this crate might still be useful. 

I made this crate when I was experimenting with [The Update Framework](https://theupdateframework.com/), which needs a way to repeatedly represent a struct of information so that cryptographic hashes of the same metadata will agree. The TUF spec recommended Canonical JSON ([see here](https://theupdateframework.github.io/specification/latest/#metaformat)) which is what I've implemented.

As far as I can tell, here are the differences:

1. RFC 8785 specifies "JSON objects MUST NOT exhibit duplicate property names", yet this crate makes no explicit check for that. Given that this is a rust crate and rust does not allow structs or maps with duplicate keys, it probably isn't a concern.

2. RFC 8785 is pretty specific about serializing unicode strings, including that certain characters must use the `\u####` notation, and certain characters must NOT use that notation, while otherwise being serialized as is. *Meanwhile* the Canonical JSON spec **makes no guarantee** that string contents can be meaningfully parsed as UTF-8, and only escapes `"` and `\`.

3. RFC 8785 specifies that Numbers (i.e. not integers) MUST be serialized in a certain fashion, meanwhile Canonical JSON *forbids* floating point Numbers, and only allows integers.

Hope this helps.
